uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/sub-controllers/toil-request.controller.js

```diff
@@ -102,29 +101,26 @@ define([
      * @return {Boolean}
      */
     function checkSubmitConditions () {
-      return !!vm.request.from_date && !!vm.request.to_date &&
-        !!vm.request.toil_duration && !!vm.request.toil_to_accrue;
+      return !!parentCtrl.request.from_date && !!parentCtrl.request.to_date &&
+        !!parentCtrl.request.toil_duration && !!parentCtrl.request.toil_to_accrue;
     }
 
     /**
      * Extends parent method. Fires calculation of expiry date when the
      * number of days changes and the expiry date can be calculated.
+     *
+     * @return {Promise}
      */
-    function changeInNoOfDays () {
-      vm._reset();
-      vm._calculateOpeningAndClosingBalance();
-
-      if (canCalculateExpiryDate()) {
-        vm.calculateToilExpiryDate();
-      }
+    function changeInNoOfDaysExtended () {
+      return canCalculateExpiryDate() ? calculateToilExpiryDate() : _.noop;
```

Comment:
Is `_.noop` really necessary? Isn't a simple
```js
return canCalculateExpiryDate() && calculateToilExpiryDate();
```
enough?

Reply:
I wonder what will happen if I do `changeInNoOfDaysExtended().then(...)` and `canCalculateExpiryDate()` returns true.

But you are right, `_.noop` is not ideal either. Changed to `$q.resolve()`

Reply:
why do you need to use `$q.resolve()` at all? I don't think you need it

This:
```js
return attemptCalculateBalanceChange().then(vm.changeInNoOfDaysExtended);
```
will simply be equivalent either to this
```js
return attemptCalculateBalanceChange().then(function () {
  return calculateToilExpiryDate()
});
```
or this
```js
return attemptCalculateBalanceChange().then(function () {
  return false;
});
```
either way it's fine

Reply:
![image](https://user-images.githubusercontent.com/3973243/33884406-6a340548-df37-11e7-9e0c-a379cd62040e.png)

`changeInNoOfDaysExtended()` says it always returns promise. If we try to use this function outside the `attemptCalculateBalanceChange` context and simply do:

```js
return changeInNoOfDaysExtended().then(...)
```
it will **not** work if `canCalculateExpiryDate()` returns `true`, since `true` is a boolean and does not have `then()` method.

My point is that if you say it returns promise - always return a promise disregarding the conditions.